sqlair.go

```diff
@@ -167,7 +109,7 @@ func (db *DB) Query(ctx context.Context, s *Statement, inputArgs ...any) *Query
 		ctx = context.Background()
 	}
 
-	sqlstmt, err := db.prepareStmt(ctx, db.sqldb, s)
+	sqlstmt, err := stmtCache.prepareStmt(ctx, db.cacheID, db.sqldb, s)
```

Comment:
I find a little bit weird that `prepareStmt` is now a method of the cache. Maybe we can call it `getOrPrepare` which is more explicit about using the cache or falling back to regular prepare.

Reply:
Agreed, I've had a look at this and refactored it to respect the abstractions better. This new way removes the need for the whole PrepareSubstrate thing which I think obfuscated the code a little. Its now more clear that the transaction prepares on a connection, not on the database since all the non-cache related logic is happening in the `Query` function.